interactive_ai/libs/iai_core_py/iai_core/services/dataset_storage_filter_service.py

```diff
@@ -100,20 +99,29 @@ def on_new_annotation_scene(
             annotation_scene_id,
         )
 
-        video: Video | None = None
-        if annotation_scene.media_identifier.media_type is MediaType.VIDEO_FRAME:
-            # Video frames are not uploaded separately, so we have to add media filter data
-            # together with the annotation scene filter data
-            video_repo = VideoRepo(dataset_storage_identifier=dataset_storage_identifier)
-            video = video_repo.get_by_id(annotation_scene.media_identifier.media_id)
+        media_id = annotation_scene.media_identifier.media_id
+        media_type = annotation_scene.media_identifier.media_type
+
+        media: Image | Video
+        match media_type:
+            case MediaType.IMAGE:
+                media = ImageRepo(dataset_storage_identifier).get_by_id(media_id)
+            case MediaType.VIDEO | MediaType.VIDEO_FRAME:
+                media = VideoRepo(dataset_storage_identifier).get_by_id(media_id)
+            case _:
+                raise ValueError(
+                    f"Cannot handle new annotation scene with media type {media_type.name}. "
+                    f"The media type should be an image or a video."
+                )
 
         dataset_storage_filter_data = DatasetStorageFilterData.create_dataset_storage_filter_data(
             media_identifier=annotation_scene.media_identifier,
-            media=video,
+            # Video frames are not uploaded separately, so we have to add media filter data
+            # together with the annotation scene filter data
+            media=media if media_type is MediaType.VIDEO_FRAME else None,
```

Comment:
Shouldn't this be 
```suggestion
            media=None if media_type is MediaType.VIDEO else media,
```

Previously images were also passing the media.